### PR TITLE
Develop

### LINKED ISSUE #114: Retrieve IOB, COB and some values aaps/loop from Nightscout
Good holidays at all,

on widget or complications, can be retrieve iob, cob, etc from nightscout?

So many aaps user groups with android watch, or when drive, ask this function, nightscout provide this parameteres, uploaded by aaps.
Maybe display this values on widget mobile or complications watch?

Thanks

### PR DESCRIPTION
**Pull request by pachi81**
Resolves #114

### CODE REVIEW

**Comment by pachi81** [comment]
@froster82 can you check the last strings:
https://github.com/pachi81/GlucoDataHandler/pull/118/files#diff-0908d478f1e1608a4d8bf0d6e84f3c23dd963a61f2ff17a7984af7e3e9f938d1
Thank you!

**Comment by pachi81** [comment]
@froster82 I'm not sure about the correct name, either LibreLink or LibreLinkUp.
But maybe only the app for LibreLink is called LibreLinkUp... I really don't know 😒

But I know that the libreview account is NOT the same as the librelinkup account. You can use the same data for both accounts.

You can also change it tomorrow and I need an additional translation for a new menu entry called "Contact".

Thank you very much!

**Comment by froster82** [comment]
@pachi81, while writing my manual for Polish users, I noticed that this part might me ambiguous.
We have the following state of affairs:
Librelink app is only used for uploading the data to a LibreView account. And this app has login and password of the main user of the sensor.
LibreLinkUp is the app used by a parent/wife. They need LibreLinkUp app, and their own LibreView account. These are also the data that need to be entered in GDH. I checked - when I entered my own LibreView login ad password (used by me to send glucodata to LV), GDH displays an error in LibreLink field. When I enter my wife's login and password to LibreView (also used by her in LibreLinkUp), everything is correct.
So in order for everything to be clear I think we need to call these features as follows:
1. LibreLinkUp follower (because the person using GDH needs to set up a 2nd LibreView account just for the purposes of receiving data via LibreLinkUp)
2. E-mail for Libreview account (of the follower using LibreLinkUp app)
3. Password for Libreview account (of the follower using LibreLinkUp app)

What do you think? If you disagree, I will revert my changes right away.
Also, I cannot find the new Contact menu entries.

**Comment by pachi81** [comment]
@froster82 I did some more changes to the strings...

Regarding you questions. Please do not use LibreView and Follower together in description, because the most people think, they can use the LibreView account data, but then they get an error (like you with your own account) and write me to help/support them ;-)
That´s why, only use "LibreLinkUp" oder "LibreLinkUp follower" (like the account of your wife).
I think your wife can not login to LibreView with these data?! (I can not test, as I use the same data for LibreView and LibreLinkUp). 
My app only support LibreLinkUp followers, which were activated in the LibreLinkUp App, after they were added in the FreeStyle Libre app....

> So in order for everything to be clear I think we need to call these features as follows:
> 1. LibreLinkUp follower (because the person using GDH needs to set up a 2nd LibreView account just for the purposes of receiving data via LibreLinkUp)
> 2. E-mail for Libreview account (of the follower using LibreLinkUp app)
> 3. Password for Libreview account (of the follower using LibreLinkUp app)
> What do you think? If you disagree, I will revert my changes right away.

Yes, as I wrote above, I disagree, as you can not use e-mail and password for you libreview account...

**Comment by froster82** [comment]
LibreLinkUp users must set their own LibreView accounts that is why I thought using the proper LibreView name would be clearer (we can log into LibreView with my wife's details, but the website is blank - waiting for data to be uploaded, so the login data work, but they direct to a "ghost" account).
But you are also right - people do not read and complain a lot :)
Let's just stick to LibreLinkUp to differentiate from the source LibreLink. I will change the Polish strings right away.

**Comment by froster82** [comment]
Thank you for your great work!
When I have you available, I have one more question regarding NS.
I myself do not use NS, so I cannot check it on my own - how do you activate NS follower? Do you need to fill in ever field in the menu on the phone? Or only some are obligatory?
One girl on the Polish forum - the one with iOS and WearOS watch, tried to activate her NS using an android phone, but failed. She could not activate the switch on the phone version after entering login data. I do not know how tech-savvy she is, and if she can cope with such issues on her own, that's why I ask.

**Review by froster82 — APPROVED** [review]

**Comment by pachi81** [comment]
@froster82 : if you like, you can get an internal test (not the open beta testing). So you can check you resources before release.
If you like, please send me an mail with playstore gmail address to my private mail: glucodatahandler@michel-inside.de.
So I can add you to the list of testers...

**Comment by froster82** [comment]
@pachi81, in the test version I was able to find a small mistake in iob/cob line.
Also, you removed the ability to disable the foreground notification - in status bar. In the previous versions it was really useful, because when someone uses a widget, the status bar notification could be easily turned off. Now if somebody does not know how to manage notifications in the app system settings, it is impossible to get rid of this notification. Can you bring back the option to disable foreground notification?

**Comment by pachi81** [comment]
> @pachi81, in the test version I was able to find a small mistake in iob/cob line. Also, you removed the ability to disable the foreground notification - in status bar. In the previous versions it was really useful, because when someone uses a widget, the status bar notification could be easily turned off. Now if somebody does not know how to manage notifications in the app system settings, it is impossible to get rid of this notification. Can you bring back the option to disable foreground notification?

Sorry, but if you disable foreground, the app may not work correct (get some feedback of people complaining...).
So I decided to make this app to run always in foreground. People who knows, what they are doing, can deactivate it, but it is their problem ;-)
I have to make this app foolproof, if it is in the playstore...
But you can still remove the content...

**Comment by froster82** [comment]
Hehe I know what you mean. 
On my phone, I use Juggluco and GDH with widget. I only have status bar notification from Juggluco. There was never any problem with GDH.
On my wife's phone, GDH works as LibrelinkUp follower. The first option for notification is switched off and she only has a widget. Also in this case there was no problem, as long as battery optimization was turned off.
For me the main reason for complaining might be the inability to fully customize the view :) You know - why the hell I keep seeing this and cannot turn it off! :)
But you are right - if somebody knows how to disable this notification, everything will be ok.
A little suggestion - maybe add a note explaining how to disable it using system app settings? This way you would reduce the number of people complaining even more :)

**Comment by pachi81** [comment]
If I add a note, how to disable, I could reenable the setting... makes no sense :-)

But if you use juggluco, you can also use my notification for delta or the trend arrow.
I use the 3 notifications, juggluco for the value and my two for arrow and delta...

**Comment by froster82** [comment]
I'm trying to be a bit of a devil's advocate here, so as to reduce the number of complaining to minimum ;)
If somebody wants to use GDH as a follower, status bar notification will - for sure- be useful. With Juggluco - this solution can also be beneficial - I see that my initial idea with two status bar icons stuck with you :D I in turn prefer your widget idea - less clutter in the status bar for me.

To sum up for now - only one string needs to be corrected in the Polish part.

**Comment by pachi81** [comment]
> I'm trying to be a bit of a devil's advocate here, so as to reduce the number of complaining to minimum ;) If somebody wants to use GDH as a follower, status bar notification will - for sure- be useful. With Juggluco - this solution can also be beneficial - I see that my initial idea with two status bar icons stuck with you :D I in turn prefer your widget idea - less clutter in the status bar for me.
> 
> To sum up for now - only one string needs to be corrected in the Polish part.

:-D No problem! Each discussion helps to make the app better. As android needs the notification for running in foreground, I need to leave it as it is. 
The next problem is, on my phone everything works well without notification, but it seems, that other guys have more problems. 
And the biggest problem, that disabling the battery optimization by popup is not wanted by google... :-(

I have found some little bug, too. I have to fix it, hopefully I have some time, today...

**Comment by pachi81** [comment]
@froster82 new test version available.
Please check, if it fits now. 
Thank you!

**Comment by pachi81** [comment]
> @pachi81, no bugs on my side. The only thing which might still be checked is the description of the Floating widget. It does not include the iob/cob info. But that is only a small interface thing. One Polish description might be still changed to better fit in context, but it does not need to be done right now.

Yes, you are right. But I also thought about, that the floating widget text should contain an info, that it can be closed by holding it for more than 5 seconds. 
But if there is no other issue, I would change it in the next version...
